pkg/alertmanager/alertmanager_metrics.go

```diff
@@ -244,8 +244,8 @@ func (m *alertmanagerMetrics) Collect(out chan<- prometheus.Metric) {
 
 	data.SendMaxOfGaugesPerUser(out, m.configHashValue, "alertmanager_config_hash")
 
-	data.SendSumOfCountersWithLabels(out, m.partialMerges, "alertmanager_partial_state_merges_total", "key")
-	data.SendSumOfCountersWithLabels(out, m.partialMergesFailed, "alertmanager_partial_state_merges_failed_total", "key")
-	data.SendSumOfCountersWithLabels(out, m.replicationTotal, "alertmanager_state_replication_total", "key")
-	data.SendSumOfCountersWithLabels(out, m.replicationFailed, "alertmanager_state_replication_failed_total", "key")
+	data.SendSumOfCountersPerUser(out, m.partialMerges, "alertmanager_partial_state_merges_total")
```

Comment:
Since you're now aggregating per user and not per key you should change the label name from `key` to `user`. It's declared above, where `partialMerges` is defined.

You should also add a CHANGELOG entry `[CHANGE]`.

_This comment applies to all other metrics you're changing too._

Reply:
I don't think we've advertised the "Sharding" operation mode of the alertmanager yet (at least, I could find reference to the metrics being added). Should we wait until the feature is roughly marked as complete before documenting? (e.g. new configuration options, new metrics)

Reply:
Oh right. If these metrics were not advertised yet, then no need for a CHANGELOG entry.